index.js

```diff
@@ -14,6 +14,11 @@ module.exports = function (input, length) {
 		return input;
 	}
 
+	var strippedTrailingSlash = input.replace(/\/$/, '');
+	if (strippedTrailingSlash.length === length) {
+		return strippedTrailingSlash;
+	}
```

Comment:
I don't think so: `http://example.com/a/b/c/` and `http://example.com/a/b/c` are different Urls.